httplib.go

```diff
@@ -54,32 +56,32 @@ func ErrorToCode(err error) int {
 // ReadError converts http error to internal error type
 // based on HTTP response code and HTTP body contents
 // if status code does not indicate error, it will return nil
-func ReadError(statusCode int, re []byte) error {
-	var e error
+func ReadError(statusCode int, respBytes []byte) error {
+	if statusCode >= http.StatusOK && statusCode < http.StatusBadRequest {
```

Comment:
Hmm, based on my read of the comment, 1xx should likely be included is non-error http codes. Although this way maintains the behaviour before the PR.